src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/header/header/Example2.java

```diff
@@ -7,6 +7,9 @@
   </module>
 </module>
 */
-
 // xdoc section -- start
+package com.puppycrawl.tools.checkstyle.checks.header.header;
+/* violation on first line 'Missing a header - not enough lines in file' */
+public class Example2 { }
```

Comment:
please create  Example4  to show violation `'Line does not match expected header` (not a 'not enough lines in file')

Please create file `java.header` in same folder with other ExampleX.java files.

please make 
`<property name="headerFile" value="${config.folder}/java.header"/>`

and defined system variable `config.folder` with value to folder with examples, in test method before execution of test, to let checkstyle to make `headerFile` expand variable and have correct path to file.
Example: https://github.com/checkstyle/checkstyle/blob/0cd3d8d7ad21c1d848d13ca32802380be147a2fe/src/test/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyCatchBlockCheckTest.java#L86-L88

Reply:
Ok, let me try